tests/testthat/test-c-api.R

```diff
@@ -174,3 +174,24 @@ test_that("r_which_operator() returns correct tokens", {
   expect_identical(which_operator(quote(`(-`(a))), "")
   expect_identical(which_operator(quote(`{-`(a))), "")
 })
+
+test_that("r_quo_get_expr() gets expression", {
+  r_quo_get_expr <- function(quo) {
+    .Call(rlang_test_r_quo_get_expr, quo)
+  }
+  r_quo_set_expr <- function(quo, expr) {
+    .Call(rlang_test_r_quo_set_expr, quo, expr)
+  }
+  r_quo_get_env <- function(quo) {
+    .Call(rlang_test_r_quo_get_env, quo)
+  }
+  r_quo_set_env <- function(quo, env) {
+    .Call(rlang_test_r_quo_set_env, quo, env)
+  }
+  quo <- quo(foo)
+
+  expect_identical(r_quo_get_expr(quo), quo_get_expr(quo))
+  expect_identical(r_quo_get_env(quo), quo_get_env(quo))
+  expect_identical(r_quo_set_expr(quo, NULL), quo_set_expr(quo, NULL))
+  expect_identical(r_quo_set_env(quo, empty_env()), quo_set_env(quo, empty_env()))
+})
```

Comment:
I think it would be good to test the C api explicitly by creating a toy package that uses it. Either by including a toy package in `rlang`, in `tests/testthat/fixtures`, or a helper function that creates such a package on the fly.

Reply:
I think it's a good idea but it is a bit involved to achieve. IIUC I need to build rlang and install it in a fake library, then build and check the embedded package in the context of that library. Since the library is not public yet as it's still in an early stage it's probably not worth the effort. I'll try to make it work at a later stage.